docs/introduction.md

```diff
@@ -1,10 +1,36 @@
 # Introduction
 
-[Wasmtime][github] is a standalone optimizing runtime for [WebAssembly],
-[the Component Model], and [WASI] by the [Bytecode Alliance][BA]. It runs
-WebAssembly code [outside of the Web], and can be used both as a command-line
-utility or as a library embedded in a larger application. Wasmtime strives to be
-a highly configurable and embeddable runtime to run on any scale of application.
+[Wasmtime][github] is a standalone runtime for WebAssembly, WASI, and the
+Component Model by the [Bytecode Alliance][BA].
+
+[WebAssembly] (abbreviated Wasm) is a binary instruction format that is designed
+to be a portable compilation target for programming languages. Wasm binaries
+typically have a `.wasm` file extension. In this documentation, we'll also use
+the textual representation of the binary files, which have a `.wat` file
+extension.
+
+[WASI] (the WebAssembly System Interface) defines interfaces that provide a
+secure and portable way to access several operating-system-like features such as
+filesystems, Berkely sockets, clocks, and random numbers.
```

Comment:
FWIW, there are low-level sockets and higher-level APIs like the HTTP proxy world. Not really worth jumping into that level of detail here, so I think just saying "networking" is good.

```suggestion
secure and portable way to access several operating-system-like features such as
filesystems, networking, clocks, and random numbers.
```